src/lib/structures/symbols.ml

```diff
@@ -317,82 +296,209 @@ let name_to_string =
+
+    (* DT theory *)
+    | Record -> Fmt.pf ppf "@Record"
+    | Access s -> Fmt.pf ppf "@Access_%a" Hstring.print s
+    | Constr s -> Hstring.print ppf s
+    | Destruct (s, g) ->
+      Fmt.pf ppf "%s%a" (if g then "" else "!") Hstring.print s
+
+    (* Float theory *)
+    | Float -> Fmt.pf ppf "float"
+    | Fixed -> Fmt.pf ppf "fixed"
+
+    | Not_theory_constant -> Fmt.pf ppf "not_theory_constant"
+    | Is_theory_constant -> Fmt.pf ppf "is_theory_constant"
+    | Linear_dependency -> Fmt.pf ppf "linear_dependency"
+
+    | Optimize {order; is_max=true} ->
+      Fmt.pf ppf "maximize(-,%d)" order
+    | Optimize {order; is_max=false} ->
+      Fmt.pf ppf "minimize(-,%d)" order
+    | Reach -> assert false
+
+  let pp_lit ppf lit =
+    match lit with
+    | L_eq -> Fmt.pf ppf "="
+    | L_neg_eq -> Fmt.pf ppf "distinct"
+    | L_built LE -> Fmt.pf ppf "<="
+    | L_built LT -> Fmt.pf ppf "<"
+    | L_neg_built LE -> Fmt.pf ppf ">"
+    | L_neg_built LT -> Fmt.pf ppf ">="
+    | L_neg_pred -> Fmt.pf ppf "not "
+    | L_built (IsConstr h) ->
+      Fmt.pf ppf "? %a" Hstring.print h
+    | L_neg_built (IsConstr h) ->
+      Fmt.pf ppf "?not? %a" Hstring.print h
+
+  let pp_form ppf f =
+    match f with
+    | F_Unit _ -> Fmt.pf ppf "/\\"
+    | F_Clause _ -> Fmt.pf ppf "\\/"
+    | F_Lemma -> Fmt.pf ppf "Lemma"
+    | F_Skolem -> Fmt.pf ppf "Skolem"
+    | F_Iff -> Fmt.pf ppf "<->"
+    | F_Xor -> Fmt.pf ppf "xor"
+
+  let pp ?(show_vars = true) ppf sy =
+    match sy with
+    | Lit lit -> pp_lit ppf lit
+    | Form form -> pp_form ppf form
+    | Let -> Fmt.pf ppf "let"
+    | Op op -> pp_operator ppf op
+
+    (* Core theory *)
+    | True -> Fmt.pf ppf "true"
+    | False -> Fmt.pf ppf "false"
+    | Void -> Fmt.pf ppf "void"
+    | Name (n, _, _) -> pp_name ppf (Hstring.view n)
+    | Var v when show_vars -> Fmt.pf ppf "'%s'" (Var.to_string v)
+    | Var v -> Fmt.string ppf (Var.to_string v)
+
+    (* Reals and Ints theories *)
+    | Int i -> Z.pp_print ppf i
+    | Real q -> Q.pp_print ppf q
+
+    (* FixedSizedBitVectors theory *)
+    | Bitv (n, s) ->
+      Fmt.pf ppf "[|%s|]" (Z.format (Fmt.str "%%0%db" n) s)
+
+    (* Symbols used in semantic triggers *)
+    | In (lb, rb) ->
+      Fmt.pf ppf "%s, %s" (string_of_bound lb) (string_of_bound rb)
+    | MapsTo v -> Fmt.pf ppf "%a |->" Var.print v
+end
+
+module SmtPrinter = struct
+  let pp_operator ppf op =
+    match op with
+    (* Core theory *)
+    |Tite -> Fmt.pf ppf "ite"
+
+    (* Reals and Ints theories *)
+    | Plus -> Fmt.pf ppf "+"
+    | Minus -> Fmt.pf ppf "-"
+    | Mult -> Fmt.pf ppf "*"
+    | Div -> Fmt.pf ppf "/"
+    | Modulo -> Fmt.pf ppf "%%"
+
+    (* Reals_Ints theory *)
+    | Abs_int | Abs_real -> Fmt.pf ppf "abs"
+    | Real_of_int -> Fmt.pf ppf "to_real"
+    | Real_is_int -> Fmt.pf ppf "is_int"
+    | Integer_round -> Fmt.pf ppf "to_int"
+
+    (* FixedSizedBitVectors theory *)
+    | Extract (i, j) ->
+      Fmt.pf ppf "(_ extract %d %d)" i j
```

Comment:
I believe that Alt-Ergo uses opposite conventions for extract arguments compared to SMT-LIB and that `i` and `j` must be flipped here (see for instance `Expr.Core.BV.extract`)